app/uk/gov/hmrc/gform/eval/ExpressionResult.scala

```diff
@@ -126,8 +126,8 @@ sealed trait ExpressionResult extends Product with Serializable {
     case t: Invalid      => false
     case t: Hidden.type  => false
     case t: Empty.type   => false
-    case t: NumberResult => false
-    case t: StringResult => false
+    case t: NumberResult => identical(er)
+    case t: StringResult => identical(er)
```

Comment:
This is incorrect, why is this change being done? Example template?

Reply:
This is to fix a failed template in the gform-acceptance
https://github.com/hmrc/gform-acceptance-tests/blob/f14a9d0175e2473cb2a5d767564403c2533959df/src/test/resources/jsonComponent/addtolist/addtolist-MiniCYAPage.json#L312

after GFORM-1714 was implented `Telephone2`, `country1` in the `includeIf` are Lists now. So we decided to use `contains` instead of `=`
but original implementation of `contains` for list 
https://github.com/hmrc/gform-frontend/blob/4c9cfbc7a655aeecbad855876787b47f3cf7e8ac/app/uk/gov/hmrc/gform/eval/ExpressionResult.scala#L139

It runs contains on each element of the list which in our case if StringResult or NumberResult. 
So instead of reimplenting `contains` for list I decided to alter implementation of `contains` for NumberResult and SringResult
Alternatively I think we can implement it like

```
case t: ListResult      => t.list.exists(_.identical(er))
```

Reply:
Isn't this fixed by #1516?

Reply:
no, it is different

Reply:
well not completely different. That change fixes the top level references to the ATL identifiers. They became Lists after the change. But `contains` on that list wont't work because `t.list.exists(_.contains(er))` and the function in `exists` always returns `false`

Reply:
I can't comprehend what you are saying. 

This looks correct to me:
```
t.list.exists(_.contains(er))
```

What you mean by
> the function in `exists` always returns `false`

BTW. I'm trying to test this and I'm failing to reproduce any case when it would not work as expected